File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    nums_len = len(nums)//2
    nums = set(nums)
    if len(nums) == nums_len:
        return len(nums)
    elif len(nums) > nums_len:
        return nums_len
    elif len(nums) < nums_len:
        return len(nums)
    return 

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        Set<Integer> pocketmons = init(nums);
        int pickCounts = nums.length / 2;
        
        if (pocketmons.size() > pickCounts) {
            return pickCounts;
        }
        return pocketmons.size();
    }
    
    private Set<Integer> init(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for  (int num : nums) {
            set.add(num);
        }
        
        return set;
    }
}